File: C/Queue.c
#include <stdio.h>

// Node struct contains an integer value and a pointer to the next Node.
typedef struct Node
{
	int value;
	struct Node* next;
} Node;

// Add new node to the end of the queue.
void enqueue(Node* head, int val)
{
	Node* cur = head;

	// Find the last node in the queue
	while (cur->next != NULL)
	{
		cur = cur->next;
	}

	// Add new node in last queue position
	cur->next = malloc(sizeof(Node));
	cur->next->value = val;
	cur->next->next = NULL;
}

// Remove the first node in the queue and return its value.
int dequeue(Node** head)
{
	// Check for empty queue
	if (*head == NULL)
	{
		return NULL;
	}
	
	// Save head node's parameters, then delete it.
	int val = (*head)->value;
	Node* newHead = (*head)->next;
	free(*head);

	// Re-assign head node pointer and return saved value.
	*head = newHead;
	return val;
}

// Print a queue in a readable format.
void printQ(Node* head)
{
	// Beginning print statement
	printf("Q(");

	// Ensure queue is not empty
	if (head != NULL)
	{
		// Print the first node in queue
		Node* cur = head;
		printf("%d", cur->value);

		// Print all subsequent nodes in queue
		while (cur->next != NULL)
		{
			cur = cur->next;
			printf(", %d", cur->value);
		}
	}

	// Ending print statement
	printf(")\n");
}

int main()
{
	// Initialize queue and test enqueue function.
	printQ(NULL);
	Node* head = malloc(sizeof(Node));
	head->value = 71;
	head->next = NULL;
	printQ(head);
	enqueue(head, 50);
	enqueue(head, 39);
	printQ(head);

	// Test dequeue function
	int a = dequeue(&head);
	int b = dequeue(&head);
	printQ(head);
	return 0;
}

File: C#/Queue.cs
using System;

namespace Structure
{
    // Nodes contain an integer value and a reference to the next node.
    class Node
    {
        public int value;
        public Node next;

        public Node(int val)
        {
            value = val;
        }
    }

    // In a queue, nodes are added to the tail and removed from the head.
    class Queue
    {
        public Node head;

        // Add a node to the tail of the queue.
        public void enqueue(int val)
        {
            // Check for empty queue
            if (head == null)
            {
                head = new Node(val);
            }

            // Find last node and make it reference the new node
            else
            {
                Node cur = head;
                while (cur.next != null)
                {
                    cur = cur.next;
                }
                cur.next = new Node(val);
            }
        }

        // Remove a node from the head of the queue and return its value.
        public int dequeue()
        {
            // Throw an exception if the queue is empty
            if (head == null)
            {
                throw new InvalidOperationException("Empty queue");
            }

            // Return the head node's value, and make the node in the
            // second position the new head node.
            int val = head.value;
            head = head.next;
            return val;
        }

        // Print a queue in a readable format.
        public void print()
        {
            // Check for empty queue
            if (head == null)
            {
                Console.WriteLine("Q()");
            }

            // Print each node's value
            else
            {
                Node cur = head;
                Console.Write("Q({0}", cur.value);
                while (cur.next != null)
                {
                    cur = cur.next;
                    Console.Write(", {0}", cur.value);
                }
                Console.Write(")\n");
            }
        }
    }

	class Program
	{
		static int Main()
		{
            // Initialize queue and test enqueue function
            Queue q = new Queue();
            q.head = new Node(8);
            q.print();
            q.enqueue(3);
            q.enqueue(2);
            q.print();

            // Test dequeue function
            int a = q.dequeue();
            q.print();
            int b = q.dequeue();
            q.print();
            int c = q.dequeue();
            q.print();
			return 0;
		}
	}
}